os/fs/smartfs/smartfs_utils.c

```diff
@@ -739,20 +739,25 @@ int smartfs_sync_internal(struct smartfs_mountpt_s *fs, struct smartfs_ofile_s *
 		if (sf->entry.datalen > 0 && used_value == 0) {
 			used_value = fs->fs_llformat.availbytes - sizeof(struct smartfs_chain_header_s);
 		}
-
 		set_used_byte_count((uint8_t *)header->used, used_value);
+
+		smartfs_setbuffer(&readwrite, sf->currsector, offsetof(struct smartfs_chain_header_s, used), sizeof(uint16_t), (uint8_t *)&fs->fs_rwbuffer[offsetof(struct smartfs_chain_header_s, used)]);
 #else
 		if (SMARTFS_USED(header) == SMARTFS_ERASEDSTATE_16BIT) {
-			header->used[0] = (uint8_t)(sf->byteswritten & 0x00FF);
-			header->used[1] = (uint8_t)(sf->byteswritten >> 8);
+			*((uint16_t *)header->used) = sf->byteswritten;
```

Comment:
actually previous is correct, if we consider MIPS....that is modified when our target board is mips, not arm based...so please roll back.. previous can support both of mips and arm based chip